src/Symfony/Component/DependencyInjection/Tests/Fixtures/FactoryDummy.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Tests\Fixtures;
+
+class FactoryDummy
+{
+    public static function createSelf(): FactoryDummy
+    {
+    }
```

Comment:
please add a test with ``self`` as return type, and another one with ``parent``, to ensure that they are resolved properly